Bio/SeqUtils/__init__.py

```diff
@@ -11,31 +11,139 @@
 
 
 import re
+import warnings
 from math import pi, sin, cos
 
 from Bio.Seq import Seq, complement, complement_rna
 from Bio.Data import IUPACData
-
+from Bio import BiopythonDeprecationWarning
 
 ######################################
 # DNA
 ######################
 # {{{
 
+_ambiguous_gc_values = {
+    "A": 0,
+    "C": 1.0,
+    "T": 0,
+    "G": 1.0,
+    "S": 1.0,
+    "V": 2 / 3,
+    "B": 2 / 3,
+    "M": 0.5,
+    "R": 0.5,
+    "Y": 0.5,
+    "K": 0.5,
+    "X": 0.5,
+    "N": 0.5,
+    "H": 1 / 3,
+    "D": 1 / 3,
+    "W": 0,
+}
+
+
+def gc_content(seq, ambiguous="remove"):
+    """Calculate G+C percentage in seq (float between 0 and 1).
+
+    Copes with mixed case sequences. Ambiguous Nucleotides in this context are
+    those different from ATCGSW.
+
+    If ambiguous equals "remove" (default), will only count GCS towards the
+    percentage, but will also only count ACTGSW towards the length of the sequence.
+    Equivalent to gc_content(seq.replace('N','')) but replacing all nucleotides
+    that are ambiguous between GC or AT (VBMRYKXNHD).
+
+    If ambiguous equals "ignore", it will treat only unambiguous
+    nucleotides (GCS) as counting towards the GC percentage.
```

Comment:
but will include all ambiguous and unambiguous nucleotides when calculating the sequence length.